### PR TITLE
Refactoring Test Base Class for Functional Tests

### PR DESCRIPTION
**Pull request by sericcson**

### CODE REVIEW

File: src/functionaltests/java/com/ericsson/ei/utils/FunctionalTestBase.java [hunk 1]
```diff
@@ -55,11 +39,17 @@
...
     private static final Logger LOGGER = LoggerFactory.getLogger(FunctionalTestBase.class);
 
+    @Autowired
+    protected EventManager eventManager;
+
+    @Autowired
+    protected DataBaseManager dbManager;
+
     @Autowired
     private MongoProperties mongoProperties;
```
Comment (evasiba-ericsson) [inline src/functionaltests/java/com/ericsson/ei/utils/FunctionalTestBase.java]: I do not think that mongo properties and mongo client are needed here.

File: src/functionaltests/java/com/ericsson/ei/utils/FunctionalTestBase.java [hunk 2]
```diff
@@ -55,13 +37,16 @@
...
 
     @Autowired
-    private MongoProperties mongoProperties;
+    protected EventManager eventManager;
+
+    @Autowired
+    protected DataBaseManager dbManager;
 
     @Autowired
     private RmqHandler rmqHandler;
```
Comment (evasiba-ericsson) [inline src/functionaltests/java/com/ericsson/ei/utils/FunctionalTestBase.java]: remove rmqHandler and waitListCollection name too

File: src/functionaltests/java/com/ericsson/ei/utils/EventManager.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import lombok.Getter;
+
+@Component
+public class EventManager {
+
+    @Autowired
+    private RmqHandler rmqHandler;
+
+    @Getter
+    private List<String> eventsIdList;
```
Comment (evasiba-ericsson) [inline src/functionaltests/java/com/ericsson/ei/utils/EventManager.java]: this field should not exist. Use only arguments.

File: src/functionaltests/java/com/ericsson/ei/utils/EventManager.java [hunk 4]
```diff
@@ -38,15 +35,24 @@
...
         List<String> eventNames = eventNamesToSend;
         JsonNode parsedJSON = getJSONFromFile(eiffelEventsJsonPath);
         for (String eventName : eventNames) {
             JsonNode eventJson = parsedJSON.get(eventName);
-            eventsIdList.add(eventJson.get("meta").get("id").toString().replaceAll("\"", ""));
             rmqHandler.publishObjectToWaitlistQueue(eventJson.toString());
         }
     }
+    
+    public List<String> getEventsIdList(String eiffelEventsJsonPath, List<String> eventNamesToSend) throws IOException {
```
Comment (Christoffer-Cortes) [inline src/functionaltests/java/com/ericsson/ei/utils/EventManager.java]: A suggestion would be to simply return the eventsIdList on the sendEiffelEvents method.

File: src/functionaltests/java/com/ericsson/ei/utils/FunctionalTestBase.java [hunk 5]
```diff
@@ -75,26 +56,6 @@
     @Value("${aggregated.collection.name}")
     private String aggregatedCollectionName;
```
Comment (evasiba-ericsson) [inline src/functionaltests/java/com/ericsson/ei/utils/FunctionalTestBase.java]: These three fields should also be removed. I did not see then previously in my view of pull request.

**Pull request merged**